
loading Tensorflow

loading scikit-learn
Lmod has detected the following error: The following module(s) are unknown:
"scikit-learn/0.23.2-fosscuda-2020b"

Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore-cache load "scikit-learn/0.23.2-fosscuda-2020b"

Also make sure that all modulefiles written in TCL start with the string
#%Module



base

Install requirements, check pandas version
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: cuda-python in /home/ddeq/.local/lib/python3.8/site-packages (from -r requirements.txt (line 1)) (12.1.0)
Requirement already satisfied: tensorboardX in /home/ddeq/.local/lib/python3.8/site-packages (from -r requirements.txt (line 2)) (2.6)
Requirement already satisfied: psutil in /opt/itu/easybuild/software/Python/3.8.2-GCCcore-9.3.0/lib/python3.8/site-packages (from -r requirements.txt (line 3)) (5.7.0)
Requirement already satisfied: opencv-python in /home/ddeq/.local/lib/python3.8/site-packages (from -r requirements.txt (line 4)) (4.2.0.32)
Requirement already satisfied: Pillow in /home/ddeq/.local/lib/python3.8/site-packages (from -r requirements.txt (line 5)) (8.3.1)
Requirement already satisfied: matplotlib in /home/ddeq/.local/lib/python3.8/site-packages (from -r requirements.txt (line 6)) (3.4.3)
Requirement already satisfied: carbontracker in /home/ddeq/.local/lib/python3.8/site-packages (from -r requirements.txt (line 7)) (1.1.7)
Collecting pandas>=2.0.0 (from -r requirements.txt (line 8))
  Using cached pandas-2.0.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.3 MB)
Requirement already satisfied: cython in /home/ddeq/.local/lib/python3.8/site-packages (from cuda-python->-r requirements.txt (line 1)) (0.29.24)
Requirement already satisfied: numpy in /opt/itu/easybuild/software/SciPy-bundle/2020.03-fosscuda-2020a-Python-3.8.2/lib/python3.8/site-packages (from tensorboardX->-r requirements.txt (line 2)) (1.18.3)
Requirement already satisfied: packaging in /home/ddeq/.local/lib/python3.8/site-packages (from tensorboardX->-r requirements.txt (line 2)) (21.0)
Requirement already satisfied: protobuf<4,>=3.8.0 in /opt/itu/easybuild/software/protobuf-python/3.13.0-fosscuda-2020a-Python-3.8.2/lib/python3.8/site-packages (from tensorboardX->-r requirements.txt (line 2)) (3.13.0)
Requirement already satisfied: cycler>=0.10 in /home/ddeq/.local/lib/python3.8/site-packages (from matplotlib->-r requirements.txt (line 6)) (0.10.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /home/ddeq/.local/lib/python3.8/site-packages (from matplotlib->-r requirements.txt (line 6)) (1.3.1)
Requirement already satisfied: pyparsing>=2.2.1 in /opt/itu/easybuild/software/Python/3.8.2-GCCcore-9.3.0/lib/python3.8/site-packages (from matplotlib->-r requirements.txt (line 6)) (2.4.7)
Requirement already satisfied: python-dateutil>=2.7 in /home/ddeq/.local/lib/python3.8/site-packages (from matplotlib->-r requirements.txt (line 6)) (2.8.2)
Requirement already satisfied: requests in /home/ddeq/.local/lib/python3.8/site-packages (from carbontracker->-r requirements.txt (line 7)) (2.26.0)
Requirement already satisfied: geocoder in /home/ddeq/.local/lib/python3.8/site-packages (from carbontracker->-r requirements.txt (line 7)) (1.38.1)
Requirement already satisfied: pynvml in /home/ddeq/.local/lib/python3.8/site-packages (from carbontracker->-r requirements.txt (line 7)) (11.5.0)
Requirement already satisfied: importlib-metadata in /home/ddeq/.local/lib/python3.8/site-packages (from carbontracker->-r requirements.txt (line 7)) (4.6.4)
Requirement already satisfied: pytz>=2020.1 in /home/ddeq/.local/lib/python3.8/site-packages (from pandas>=2.0.0->-r requirements.txt (line 8)) (2023.3)
Requirement already satisfied: tzdata>=2022.1 in /home/ddeq/.local/lib/python3.8/site-packages (from pandas>=2.0.0->-r requirements.txt (line 8)) (2023.3)
Collecting numpy (from tensorboardX->-r requirements.txt (line 2))
  Using cached numpy-1.24.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
Requirement already satisfied: six in /home/ddeq/.local/lib/python3.8/site-packages (from cycler>=0.10->matplotlib->-r requirements.txt (line 6)) (1.16.0)
Requirement already satisfied: setuptools in /opt/itu/easybuild/software/Python/3.8.2-GCCcore-9.3.0/lib/python3.8/site-packages (from protobuf<4,>=3.8.0->tensorboardX->-r requirements.txt (line 2)) (45.2.0)
Requirement already satisfied: click in /home/ddeq/.local/lib/python3.8/site-packages (from geocoder->carbontracker->-r requirements.txt (line 7)) (8.0.1)
Requirement already satisfied: future in /opt/itu/easybuild/software/Python/3.8.2-GCCcore-9.3.0/lib/python3.8/site-packages (from geocoder->carbontracker->-r requirements.txt (line 7)) (0.18.2)
Requirement already satisfied: ratelim in /home/ddeq/.local/lib/python3.8/site-packages (from geocoder->carbontracker->-r requirements.txt (line 7)) (0.1.6)
Requirement already satisfied: zipp>=0.5 in /home/ddeq/.local/lib/python3.8/site-packages (from importlib-metadata->carbontracker->-r requirements.txt (line 7)) (3.5.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/ddeq/.local/lib/python3.8/site-packages (from requests->carbontracker->-r requirements.txt (line 7)) (1.26.6)
Requirement already satisfied: certifi>=2017.4.17 in /home/ddeq/.local/lib/python3.8/site-packages (from requests->carbontracker->-r requirements.txt (line 7)) (2021.5.30)
Requirement already satisfied: charset-normalizer~=2.0.0 in /home/ddeq/.local/lib/python3.8/site-packages (from requests->carbontracker->-r requirements.txt (line 7)) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /home/ddeq/.local/lib/python3.8/site-packages (from requests->carbontracker->-r requirements.txt (line 7)) (3.2)
Requirement already satisfied: decorator in /opt/itu/easybuild/software/Python/3.8.2-GCCcore-9.3.0/lib/python3.8/site-packages (from ratelim->geocoder->carbontracker->-r requirements.txt (line 7)) (4.4.2)
Installing collected packages: numpy, pandas
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
albumentations 1.0.0 requires PyYAML, which is not installed.
albumentations 1.0.0 requires scikit-image>=0.16.1, which is not installed.
imgaug 0.4.0 requires imageio, which is not installed.
imgaug 0.4.0 requires scikit-image>=0.14.2, which is not installed.
tensorflow 2.3.1 requires numpy<1.19.0,>=1.16.0, but you have numpy 1.24.2 which is incompatible.
Successfully installed numpy-1.21.2 pandas-2.0.0
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done


==> WARNING: A newer version of conda exists. <==
  current version: 4.10.1
  latest version: 23.3.1

Please update conda by running

    $ conda update -n base -c defaults conda



# All requested packages already installed.

Pandas version: 1.0.3

Making thyroid dataframe
2023-04-18 21:57:03.497586: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
Traceback (most recent call last):
  File "src/make_dataframe.py", line 14, in <module>
    dataframe = collect_data(home, args.dataset)
  File "/home/data_shares/ccdd/revisiting-transfer/src/io_fun/data_import.py", line 474, in collect_data
    dataframe = import_thyroid(data_dir)
  File "/home/data_shares/ccdd/revisiting-transfer/src/io_fun/data_import.py", line 271, in import_thyroid
    meta_df = pd.read_xml(file_path)
  File "/opt/itu/easybuild/software/SciPy-bundle/2020.03-fosscuda-2020a-Python-3.8.2/lib/python3.8/site-packages/pandas/__init__.py", line 263, in __getattr__
    raise AttributeError(f"module 'pandas' has no attribute '{name}'")
AttributeError: module 'pandas' has no attribute 'read_xml'
